core/templates/dev/head/filters.js

```diff
@@ -341,26 +341,45 @@ oppia.filter('normalizeWhitespacePunctuationAndCase', [function() {
   };
 }]);
 
-// Note that this filter removes additional new lines or <p><br></p> tags
-// at the end of the string.
-oppia.filter('removeExtraLines', [function() {
+// Filter that targets to remove occurance of blank lines, empty paragraphs
+// and spaces at the end of string. It'll remove <p></p>, <p><br></p>, spaces
+// and &nbsp; occurances from the end of the string.
+oppia.filter('removeExtraLinesAndSpace', [function() {
   return function(string) {
     if (!angular.isString(string)) {
       return string;
     }
     var BLANK_LINES_TEXT = '<p><br></p>';
     var EMPTY_PARA_TEXT = '<p></p>';
     while (1) {
-      var endIndex = string.length;
-      var bStr = string.substring(endIndex - BLANK_LINES_TEXT.length, endIndex);
-      var pStr = string.substring(endIndex - EMPTY_PARA_TEXT.length, endIndex);
-      if (bStr === BLANK_LINES_TEXT) {
-        string = string.substring(0, endIndex - BLANK_LINES_TEXT.length);
-      } else if (pStr === EMPTY_PARA_TEXT) {
-        string = string.substring(0, endIndex - EMPTY_PARA_TEXT.length);
-      } else {
+      var strlength = string.lastIndexOf('<');
+      if (string[strlength - 1] !== ' ' &&
+          string.substring(strlength - 4, strlength) !== '<br>' &&
+          string.substring(strlength - 6, strlength) !== '&nbsp;') {
         break;
       }
+      while (1) {
+        var lng = string.lastIndexOf('<');
```

Comment:
Why is this named differently from strlength?

Reply:
I guess that's not needed. Forgot to remove after I changed my approach to tackle. Done